openforcefield/tests/test_forcefield.py

```diff
@@ -3291,21 +3295,70 @@ def test_parameterize_protein(self, toolkit_registry, registry_description):
+        mol = create_ethanol()
+        mol.assign_fractional_bond_orders(bond_order_model="am1-wiberg")
+        top = Topology.from_molecules(mol)
+
+        mod_mol = create_ethanol()
+        mod_mol.assign_fractional_bond_orders(bond_order_model="am1-wiberg")
+        # populate the mol with garbage bond orders
+        for bond in mod_mol.bonds:
+            bond.fractional_bond_order = 3 - bond.fractional_bond_order
+        mod_top = Topology.from_molecules(mod_mol)
+
+        default_bo = [
+            b.fractional_bond_order for b in [*top.reference_molecules][0].bonds
+        ]
+        mod_bo = [
+            b.fractional_bond_order for b in [*mod_top.reference_molecules][0].bonds
+        ]
+        assert not default_bo == mod_bo
+
+        forcefield = ForceField("test_forcefields/smirnoff99Frosst.offxml", xml_ff_bo)
+
+        omm_system = forcefield.create_openmm_system(top)
+        mod_omm_system = forcefield.create_openmm_system(mod_top)
+
+        default_bond_force = [
+            f for f in omm_system.getForces() if isinstance(f, openmm.HarmonicBondForce)
+        ][0]
+        mod_bond_force = [
+            f
+            for f in mod_omm_system.getForces()
+            if isinstance(f, openmm.HarmonicBondForce)
+        ][0]
+
+        for idx in range(default_bond_force.getNumBonds()):
+            assert default_bond_force.getBondParameters(
+                idx
+            ) == mod_bond_force.getBondParameters(idx)
+
+    @requires_openeye
     @pytest.mark.parametrize(
-        ("get_molecule", "k_interpolated", "central_atoms"),
+        (
+            "get_molecule",
+            "k_torsion_interpolated",
+            "k_bond_interpolated",
+            "central_atoms",
+        ),
         [
-            (create_ethanol, 4.953856, (1, 2)),
-            (create_reversed_ethanol, 4.953856, (7, 6)),
+            (create_ethanol, 4.953856, 42208.5401, (1, 2)),
+            (create_reversed_ethanol, 4.953856, 42208.5401, (7, 6)),
         ],
     )
-    def test_fractional_bondorder(self, get_molecule, k_interpolated, central_atoms):
-        """Test the fractional bond orders are used to interpolate k values as we expect"""
+    def test_fractional_bondorder(
+        self, get_molecule, k_torsion_interpolated, k_bond_interpolated, central_atoms
+    ):
+        """Test the fractional bond orders are used to interpolate k values as we expect
+        Until toolkit-specific differences in fractional bond order generation (for harmonic bond forces) are handled,
+        this test is pinned to using OpenEye."""
 
         mol = get_molecule()
 
-        forcefield = ForceField(
-            "test_forcefields/smirnoff99Frosst.offxml", xml_ff_torsion_bo
-        )
+        forcefield = ForceField("test_forcefields/smirnoff99Frosst.offxml", xml_ff_bo)
         topology = Topology.from_molecules(mol)
 
         omm_system = forcefield.create_openmm_system(
```

Comment:
If we choose to make Bonds respect `partial_bond_orders_from_molecules` (I think we should), we'll want to test that pre-specified bond orders are respected by the bond parameter assignment as they are for proper torsions.